### ISSUE TITLE
Resource metadata item value truncated despite ample space in workspace dashboard table

### ISSUE THREAD

**Issue by kconley-sq** [coder/coder#8887]
In Coder v0.27.3 we're seeing that the value of a resource metadata item is truncated with ellipsis even though there seems to be enough space to show more (or even all) of the value in the workspace dashboard table on the Coder website:

<img width="1230" alt="image" src="https://github.com/coder/coder/assets/93007527/52f1c439-83c3-4641-a309-8f2ff0be7b10">

```terraform
resource "coder_metadata" "device" {
  # other fields elided...
  item {
    key   = "Test"
    value = "The quick brown fox jumped over the lazy dog"
  }
}
```

Can that be improved to show more (or all if possible) of the value?

**Comment by matifali** [comment]
cc: @BrunoQuaresma

**Comment by BrunoQuaresma** [comment]
More context about this, since we don't control what kind of information the user can add to it, it is hard to control and guess what is the best width so we are setting it statically using a fixed width. We could also re-think the design of it if that is the case.

**Comment by matifali** [comment]
@BrunoQuaresma what do you think about having a field `width` in coder metadata blocks. So that the user can set a custom width for their agent metadata. 
```hcl
resource "coder_metadata" "device" {
  # other fields elided...
  item {
    key   = "Test"
    value = "The quick brown fox jumped over the lazy dog"
    width = "50px" 
  }
}
```

**Comment by BrunoQuaresma** [comment]
It looks hacky but at the same time an available option 🤔 I would like to have more opinions on this.

**Comment by kconley-sq** [comment]
I'm hoping template developers might not have to specify `width` manually, I'd prefer to let the Coder website UI decide how best to present the string based on the actual string length received from the agent and the user's screen resolution, device type, etc.

It looks like the Coder website UI has available space to show more of the string in this issue as well as the multiple metadata strings in #9457, so as a first step would it be possible to make the UI show as many metadata strings (and as much of those strings) as possible given the available space in the agent row on the dashboard?

**Comment by aslilac** [comment]
I spent a bunch of time trying to get something "elegant" working and came up quite short. `flex` and `grid` are both a pain to work with. One idea I had though that should improve this without too much jank: what if we only enable to grid rules if there is more than one item? I'll play around with that on Friday and see how it feels.

cc @BrunoQuaresma

**Comment by BrunoQuaresma** [comment]
@aslilac since you are thinking about this, another "side effect" of this issue is this one: https://github.com/coder/coder/issues/9457

**Comment by BrunoQuaresma** [comment]
@aslilac can I assign this to me? I have an idea how to make that happen if that is ok for you 👀

**Comment by aslilac** [comment]
> @aslilac can I assign this to me? I have an idea how to make that happen if that is ok for you 👀

yes __please__ 😂

**Comment by aslilac** [comment]
@BrunoQuaresma I'll let you reopen and reassign if you still wanna do further work on this!

**Comment by kconley-sq** [comment]
I understand the changes in #9832 to help with this issue only take effect if you only have one metadata item for a resource, but unfortunately I have two. 😅 Can we reopen this ticket to track fixing this issue independent of the number of metadata items used?

**Comment by kconley-sq** [comment]
> I understand the changes in #9832 to help with this issue only take effect if you only have one metadata item for a resource, but unfortunately I have two. 😅 Can we reopen this ticket to track fixing this issue independent of the number of metadata items used?

cc @bpmct, can we reopen this issue? I can still reproduce [the scenario described in the first comment on the issue](https://github.com/coder/coder/issues/8887#issue-1835645057) (where two metadata items are used) in v2.4.0.

**Comment by BrunoQuaresma** [comment]
@kconley-sq we are working on a new workspace page that is probably going to fix that. It is probably going to be available as an experiment around January. Does it work?

**Comment by kconley-sq** [comment]
> @kconley-sq we are working on a new workspace page that is probably going to fix that. It is probably going to be available as an experiment around January. Does it work?

Yes that sounds good, thank you @BrunoQuaresma!

**Comment by kconley-sq** [comment]
@BrunoQuaresma sure, here is an example:
```
key1: xxxxxxxxxxxx
value1: xx

key2: xxxxxxxxxxxxxxxxxxxxxx
value2: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```

**Comment by BrunoQuaresma** [comment]
@kconley-sq I think this is solved on 2.7.0 but feel free to re-open it in case it is not 🙏
